gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTooFarFromTripShapeValidatorTest.java

```diff
@@ -0,0 +1,171 @@
+    public static GtfsStop createStop(
+            long csvRowNumber,
+            String stopId,
+            double stopLat,
+            double stopLon,
+            GtfsLocationType locationType) {
+        return new GtfsStop.Builder()
+                .setCsvRowNumber(csvRowNumber)
+                .setStopId(stopId)
+                .setStopLat(stopLat)
+                .setStopLon(stopLon)
+                .setLocationType(locationType.getNumber())
+                .build();
+    }
+
+    public static GtfsStopTime createStopTime(
+            long csvRowNumber, String tripId, String stopId, int stopSequence) {
+        return new GtfsStopTime.Builder()
+                .setCsvRowNumber(csvRowNumber)
+                .setTripId(tripId)
+                .setStopId(stopId)
+                .setStopSequence(stopSequence)
+                .build();
+    }
+
+    public static GtfsShape createShape(
+            long csvRowNumber,
+            String shapeId,
+            double shapePtLat,
+            double shapePtLon,
+            int shapePtSequence) {
+        return new GtfsShape.Builder()
+                .setCsvRowNumber(csvRowNumber)
+                .setShapeId(shapeId)
+                .setShapePtLat(shapePtLat)
+                .setShapePtLon(shapePtLon)
+                .setShapePtSequence(shapePtSequence)
+                .build();
+    }
+
+    public static GtfsTrip createTrip(long csvRowNumber, String tripId, String shapeId) {
+        return new GtfsTrip.Builder()
+                .setCsvRowNumber(csvRowNumber)
+                .setTripId(tripId)
+                .setShapeId(shapeId)
+                .build();
+    }
+
+    @Test
+    public void stopWithinTripShapeShouldNotGenerateNotice() {
+        // See map of trip shape and stops (in GeoJSON) at
+        // https://gist.github.com/barbeau/d9c0b90a26a3e2ba105cae5f0e8aec4a
+        final NoticeContainer noticeContainer = new NoticeContainer();
+        StopTooFarFromTripShapeValidator validator = new StopTooFarFromTripShapeValidator();
+
+        // Create stopTable:
+        List<GtfsStop> stops = new ArrayList<>();
+        stops.add(
+                createStop(1, "1001", 28.05811731042478d, -82.41616877502503d, GtfsLocationType.STOP));
+        stops.add(
+                createStop(2, "1002", 28.05812364854794d, -82.41617370439423d, GtfsLocationType.STOP));
+        validator.stopTable = GtfsStopTableContainer.forEntities(stops, noticeContainer);
+
+        // Create stopTimeTable:
+        List<GtfsStopTime> stopTimes = new ArrayList<>();
+        stopTimes.add(createStopTime(1, "trip1", "1001", 1));
+        stopTimes.add(createStopTime(2, "trip1", "1002", 2));
+        validator.stopTimeTable = GtfsStopTimeTableContainer.forEntities(stopTimes, noticeContainer);
+
+        // Create shapeTable:
+        List<GtfsShape> shapes = new ArrayList<>();
+        shapes.add(createShape(1, "shape1", 28.05724310653972d, -82.41350776611507d, 1));
```

Comment:
shapes.add(GtfsShape.Builder()
                .setCsvRowNumber(1)
                .setShapeId("shape1")
                .setShapePtLat(28.05746701492806)
                .setShapePtLon(-82.41493135129478)
                .setShapePtSequence(2)
                .build());

Reply:
Yes. I have just adjusted to use builders in test methods directly, rather than having extra builder helper methods.